lmos-runtime-core/src/main/kotlin/ai/ancf/lmos/runtime/outbound/LmosOperatorAgentRegistry.kt

```diff
@@ -55,7 +61,22 @@ class LmosOperatorAgentRegistry(private val lmosRuntimeProperties: LmosRuntimePr
         channelId: String,
         subset: String?,
```

Comment:
runtime shouldn't know anything about the subset concept.
Istio will take care of adding the header.

Reply:
subset is to make it work like sticky session for a conversation, so that the end user is exposed to same set of capabilities for agent.
Initially i.e. for the first turn of a conversation, istio will add it. however for subsequent turn same subset should be used

Reply:
But this implies that there could be multiple routines cached for a single conversation. Why is the conversationId not sufficient for a caching key?

Reply:
If the conversationId is used as the cache key, multiple requests will be made to the operator to fetch the routing information for the same tenant and channel (different conversationId), even though the routing is specific to the tenant and channel, not the conversationId.

Reply:
Yes, but this is the plan. Otherwise we couldn't move customers gradually to a canary version. A routing is not specific for all conversations of a tenant/channel. We use Istio to pick a subset on a percentage rollout strategy.

Reply:
And yes, the operator is called once for every new conversation

Reply:
Updated the code to gradually move customers to the canary version. Since the subset is null at the start of a conversation, routing info is fetched from the operator. Also, using channel and subset as cache keys to reduce cached data compared to using conversationId.

Reply:
Please just remove the subset. The Runtime should not know anything about the subset.

Reply:
done, please check